Source/carthage/Formatting.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Formatting.swift
+//  Carthage
+//
+//  Created by J.D. Healy on 1/29/15.
+//  Copyright (c) 2015 Carthage. All rights reserved.
+//
+
+import Commandant
+import Foundation
+import LlamaKit
+import PrettyColors
+import ReactiveCocoa
+
+/// Wraps or passes through a string.
+func wrap(colorful: Bool)(wrap: Color.Wrap)(string: String) -> String {
+	return colorful ? wrap.wrap(string) : string
+}
+
+/// Information about the possible parent terminal.
+internal struct Terminal {
+	static var term: String? {
+		return getEnvironmentVariable("TERM").value()
+	}
+	static var isDumb: Bool {
+		return (Terminal.term?.lowercaseString as NSString?)?.isEqualToString("dumb") ?? false
+	}
+	static var isTTY: Bool {
+		return isatty(STDOUT_FILENO) == 1
+	}
+}
+
+public enum ColorArgument: String, ArgumentType, Printable {
+	case Auto = "auto"
+	case Never = "never"
+	case Always = "always"
+	
+	/// Whether to color and format.
+	public var isColorful: Bool {
+		switch self {
+		case .Always:
+			return true
+		case .Never:
+			return false
+		case .Auto:
+			return Terminal.isTTY && !Terminal.isDumb
+		}
+	}
+	
+	public var description: String {
+		return self.rawValue
+	}
+	
+	public static let name = "color"
+	
+	public static func fromString(string: String) -> ColorArgument? {
+		return self(rawValue: string.lowercaseString)
+	}
+	
+}
+
+public struct ColorOptions: OptionsType {
+	let argument: ColorArgument
+	let formatting: Formatting
+	
+	struct Formatting {
```

Comment:
Can this either be `public` or `private`? I like having the explicit indicator of how this is supposed to be used.